src/app/landing/shared/services/repository.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, OnDestroy } from "@angular/core";
+import { CoursesByTechnology, CoursesRepository } from "../interfaces/course";
+import { mockCourses } from "./mockCourses";
+import { Observable, Subject, delay, of, retry, shareReplay, takeUntil } from "rxjs";
+import { mockStutendsFeedback } from "./mockStudentsFeedback";
+import { StudentFeedback, StudentFeedbacksRepository } from "../interfaces/student-feedback";
+import { Mentor, MentorsRepository } from "../interfaces/mentor";
+import { mockMentors } from "./mockMentors";
+
+@Injectable({providedIn: 'root'})
+export class RepositoryService implements 
+    CoursesRepository, 
+    StudentFeedbacksRepository, 
+    MentorsRepository,
+    OnDestroy
+{
+    private coursesCache: Observable<CoursesByTechnology> | null = null
+    private studentsFeedbackCache: Observable<StudentFeedback[]> | null = null
+    private mentorsCache: Observable<Mentor[]> | null = null
+
+    private unsubscribe$ = new Subject()
+
+    private prepareAPIRequest<T>(request: Observable<T>): Observable<T>{
+        return request.pipe(
+            retry({delay: 1000}),
```

Comment:
Check how this construction with retry actually works on errors (mock it for test only).

It seems like you will not get desired result. Retry with delay only recall observable unlimited number of time if will not be success result

Reply:
Indented behavior was that if internet connection will become unavailable and then, for example, you switch to landing page it will keep trying until it will become available again. I just not handling any specific error like incorrect url or server error so in this case it will also keep retrying.

Reply:
I did test. Only thing to note is that ```_getRequest()``` function is called any time when ```getRequest()``` is called, but it just return cold observable, and only first one is saved and other that follows are ignored. Everything is pretty much as i expected, because i assume here that lending don't change for months. 

I'm going to split this into component service layer and API call layer as you suggested below, should i resolve this conversation or just start pushing new commits into this pull request?

Reply:
@MyColaAnikeiev yes, let's move the all changes inside `repositoryService` to separate PR.

Also, you can try to check the program behavior without  `prepareAPIRequest` method (the result should be the same).

This is a redundant method for the current implementation. There isn't a caching implementation, and the `retry` operator isn't used because you use mock data now (and there aren't server API calls). 
For example, let's imagine the next user case:
1. User opens the landing page (load all necessary data from the server)
2. The user goes to other pages. At this time the server has updated the `courses` list.
3. The user returns to the landing page, and there is still an "old" `courses` list.
The cache functionality isn't a trivial task. If it (cache) is not a system requirement I recommend missing this feature (you always can do the implementation of it in the future).

Also, you don't handle HTTP errors ("server unavailable", for ex).

So, let's move this part to the next pack of changes.